python/tk_desktop/desktop_window.py

```diff
@@ -947,12 +944,10 @@ def _on_different_user(self, site, user_id):
 
             dialog = BrowserIntegrationUserSwitchDialog(msg, self)
 
-            # The following applies to macOS only and has no side-effect on other plaforms.
-            # If the dialog is pinned, it means it is also in background. We'll bring the app to the foreground
-            # so keyboard focus is granted automatically to the BrowserIntegrationUserSwitchDialog instead
-            # of being unfocussed.
-            if self.is_pinned():
-                osutils.make_app_foreground()
+            # The following applies to macOS only and has no side-effect on other platforms.
```

Comment:
Seems like in PySide2 we need to always make this call now. As for making this call in PySide1, this code has no noticeable side-effect when the dialog is not pinned, so I've simplified the code.